### PR TITLE
[Tests] AsyncSubtensor (Part 1)

### PR DESCRIPTION
**Pull request by basfroman**

### CODE REVIEW

File: bittensor/core/async_subtensor.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ async def get_subnets(self, block_hash: Optional[str] = None) -> list[int]:
         return (
             []
             if result is None or not hasattr(result, "records")
-            else [netuid async for netuid, exists in result if exists]
+            else [netuid for netuid, exists in result.records if exists]
```
Comment (thewhaleking) [inline bittensor/core/async_subtensor.py]: This won't work.

File: tests/unit_tests/test_async_subtensor.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    assert mocked_decode_account_id.mock_calls == [
+        mocker.call("0"),
+        mocker.call("1"),
+        mocker.call("2"),
+        mocker.call("3"),
+    ]
+
+
+@pytest.mark.asyncio
+async def test_encode_params(subtensor, mocker):
```
Comment (thewhaleking) [inline tests/unit_tests/test_async_subtensor.py:47]: I'm not sure what this actually tests. I would think it should fail for invalid types, but it doesn't.

File: tests/unit_tests/test_async_subtensor.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    )
+    subtensor.get_netuids_for_hotkey = mocked_get_netuids_for_hotkey
+
+    # Call
+    result = await subtensor.is_hotkey_registered_any(
+        hotkey_ss58="hotkey", block_hash="FAKE_HASH"
+    )
+
+    # Asserts
+    assert result == (len(mocked_get_netuids_for_hotkey.return_value) > 0)
```
Comment (thewhaleking) [inline tests/unit_tests/test_async_subtensor.py]:
```suggestion
    assert result is (len(mocked_get_netuids_for_hotkey.return_value) > 0)
```

Use `is` for bool comparison

File: tests/unit_tests/test_async_subtensor.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    result = await subtensor.get_subnets(block_hash=fake_block_hash)
+
+    # Asserts
+    mocked_substrate_query_map.assert_called_once_with(
+        module="SubtensorModule",
+        storage_function="NetworksAdded",
+        block_hash=fake_block_hash,
+        reuse_block_hash=True,
+    )
+    assert result == result
```
Comment (thewhaleking) [inline tests/unit_tests/test_async_subtensor.py]: I don't think this is what you intended

**Review by thewhaleking — CHANGES_REQUESTED** [review]

File: bittensor/core/async_subtensor.py (hunk 1 above)
Comment (basfroman) [reply bittensor/core/async_subtensor.py]: It works perfect. Let's try yourself. But updated.

File: tests/unit_tests/test_async_subtensor.py (hunk 2 above)
Comment (basfroman) [reply tests/unit_tests/test_async_subtensor.py:47]: As we have already discussed, this tests the actual calls internally. The error case is tested in a subsequent test.

**Review by thewhaleking — APPROVED** [review]

**Pull request merged**